flash/image/embedding/strategies/default.py

```
@@ -0,0 +1,86 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+from typing import Any, Optional
+
+import torch
+
+from flash.core.adapter import Adapter, AdapterTask
+from flash.core.data.io.input import DataKeys
+from flash.core.model import Task
+from flash.core.registry import FlashRegistry
+from flash.core.utilities.url_error import catch_url_error
+
+
+class DefaultAdapter(Adapter):
+    """The ``DefaultAdapter`` is an :class:`~flash.core.adapter.Adapter`."""
+
+    required_extras: str = "image"
+
+    def __init__(self, backbone: torch.nn.Module):
+        super().__init__()
+
+        self.backbone = backbone
+
+    @classmethod
+    @catch_url_error
+    def from_task(
+        cls,
+        *args,
+        task: AdapterTask,
+        backbone: torch.nn.Module,
+        **kwargs,
+    ) -> Adapter:
+        adapter = cls(backbone)
+        adapter.__dict__["_task"] = task
+        return adapter
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        raise NotImplementedError(
+            "Training a `ImageEmbedder` with `DefaultAdapter` is not supported. Use a different strategy instead."
+        )
+
+    def validation_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DataKeys.INPUT], batch[DataKeys.TARGET])
```

Let's raise an error for validation and test too :smiley: